R/modules.R

```diff
@@ -321,11 +321,124 @@ modules <- function(..., label = "root") {
 # printing methods ----
 
 #' @rdname teal_modules
+#' @param what character vector specifying which metadata to display. 
+#'   Possible values: "data_sources", "properties", "ui_args", "server_args", "transformers"
 #' @export
-format.teal_module <- function(x, indent = 0, ...) {
-  paste0(paste(rep(" ", indent), collapse = ""), "+ ", x$label, "\n", collapse = "")
+format.teal_module <- function(x, indent = 0, is_last = FALSE, parent_prefix = "", what = c("data_sources", "properties", "ui_args", "server_args", "transformers"), ...) {
+  empty_text <- "─ none"
+  branch <- if (is_last) "└─" else "├─"
+  current_prefix <- paste0(parent_prefix, branch, " ")
+  content_prefix <- paste0(parent_prefix, if (is_last) "   " else "│  ")
+
+  format_list <- function(lst, empty = empty_text) {
+    if (is.null(lst) || length(lst) == 0) {
+      empty
+    } else {
+      paste(names(lst), collapse = ", ")
+    }
+  }
+
+  bookmarkable <- isTRUE(attr(x, "teal_bookmarkable"))
+  reportable <- "reporter" %in% names(formals(x$server))
+
+  transformers <- if (length(x$transformers) > 0) {
+    paste(sapply(x$transformers, function(t) attr(t, "label")), collapse = ", ")
+  } else {
+    empty_text
+  }
```

Comment:
Any way we can go level deeper and provide deeper print for transformers? They also have datanames and ui/server args?

Reply:
I think we have to implement format for `teal_transform_module` and just use it over here.
But I'm skeptical a little. Perhaps it should not be printed by default and only when explicitly asked by the user.

Reply:
Sure, I can understand some users would like to see in-depth overview and some just a glimpse. Would you be able to provide such parameter. We could name it `transformer` and have options `simple` and `full`?

Reply:
I think we can circle back to this when we work on teal transform module print method,

Reply:
Do we have an issue for a print method for teal transform module? Can you put the link to this conversation in that issue, so we don't forget, once we start writing print for teal transform module?

Reply:
Issue created #1403